sdk/filterutils/filter_result.go

```diff
@@ -0,0 +1,6 @@
+package filterutils
+
+type FilterResult struct {
```

Comment:
Can we simply move FilterResult to filter.go?

Reply:
This was leading to circular dependency as this would be needed by mocks as well.

Reply:
Why would it as this type has no dependency to anything? Maybe I'm missing something.

Reply:
Correct, but the module of the type needs to imported which will import other things.

Reply:
`FilterResult` is needed by mock and we use `Mock` in `Multifilter_test` (which is part of filter package)  so if we put `FilterResult` in `filter.go` package we would have a circular dependency.

Either we move `FilterResult` to different package, or we have to move `Multifilter` to separate package.